core/src/broadcast_stage.rs

```diff
@@ -86,7 +86,7 @@ impl BroadcastStageType {
         blockstore: Arc<Blockstore>,
         bank_forks: Arc<RwLock<BankForks>>,
         shred_version: u16,
-        shred_receiver_addr: Option<SocketAddr>,
+        shred_receiver_addr: Arc<RwLock<Option<SocketAddr>>>,
```

Comment:
can we use Option<Arc<RwLock<SocketAddr>> here? doing Arc::new(RwLock::new(None)) is dirty

Reply:
ah nvmd, you want to be able to turn it on + off dynamically. will leave comment here in case someone else gets confused :)

Reply:
should we just use a mutex tho?

Reply:
looks like it's written to infrequently, and read frequently in 2 different threads (broadcast_stage, and retransmit_stage) 

so, i think it should stay, but not 100% sure how often the reads are hit

Reply:
ah yeah great point. lets keep as is